File: android/app/src/main/java/com/belldien/pulseboard/ui/PulseViewModel.kt
package com.belldien.pulseboard.ui

import androidx.lifecycle.ViewModel
import com.belldien.pulseboard.data.UserRepository
import com.belldien.pulseboard.domain.*
import com.belldien.pulseboard.model.*
import kotlinx.coroutines.flow.MutableStateFlow
import kotlinx.coroutines.flow.StateFlow

class PulseViewModel : ViewModel() {
    private val repository = UserRepository()
    private val loadUsers = LoadUsersUseCase(repository)
    private val loadDetail = LoadUserDetailUseCase(repository)
    private val _home = MutableStateFlow(HomeUiState(users = loadUsers(), loading = false))
    val home: StateFlow<HomeUiState> = _home

    fun detail(id: Int): DetailUiState =
        loadDetail(id)?.let { DetailUiState(it, false) }
            ?: DetailUiState(loading = false, error = "没有找到这位朋友")
}

File: ios/PulseBoard/ViewModel/PulseViewModel.swift
import Combine
import Foundation

@MainActor final class PulseViewModel: ObservableObject {
  @Published private(set) var homeState = HomeUIState()
  private let loadUsers: LoadUsersUseCase
  private let loadDetail: LoadUserDetailUseCase
  init(repository: UserRepositoryProtocol = UserRepository()) {
    loadUsers = LoadUsersUseCase(repository: repository)
    loadDetail = LoadUserDetailUseCase(repository: repository)
    homeState = HomeUIState(users: loadUsers.execute(), isLoading: false)
  }
  func detail(for id: Int) -> DetailUIState {
    guard let user = loadDetail.execute(id: id) else {
      return DetailUIState(user: nil, isLoading: false, errorMessage: "没有找到这位朋友")
    }
    return DetailUIState(user: user, isLoading: false)
  }
}
